src/line/i-chartBuilder.test.js

```diff
@@ -0,0 +1,400 @@
+/**
+ * @jest-environment jsdom
+ */
+
+const fs = require("fs")
+const domTesting = require('@testing-library/dom')
+require('@testing-library/jest-dom')
+const userEvent = require("@testing-library/user-event").default
+
+//Spy on
+const window = require("./window")
+ 
+function initDomFromFiles(htmlPath, jsPath) {
+    window.localStorage.clear()
+    const html = fs.readFileSync(htmlPath, 'utf8')
+    document.open()
+    document.write(html)
+    document.close()
+    jest.isolateModules(function() {
+        require(jsPath)
+    })
+}
+
+/*
+    They should verify the behaviors described above. Note that these behaviors are 
+    duplicated across the three chart builder pages (bar.html, line.html, and scatter.html). 
+    You only need to test these behaviors on one of those pages.
+*/
+
+//ADDING VALUES TO THE CHART BUILDER
+
+test('Pressing "+" adds exactly one empty x and y value box', async function(){
+    initDomFromFiles(`${__dirname}/line.html`, `${__dirname}/line.js`)
+
+    //get + button for adding  new x and y cells.
+    const buttons = domTesting.queryAllByRole(document, "button")
+    const addButton = buttons[3];
+    
+    //make sure the correct button is selected
+    expect(addButton).toHaveTextContent("+")
+
+    //count all cells
+    var xValues = await domTesting.getAllByLabelText(document, "X")
+    var yValues = await domTesting.getAllByLabelText(document, "Y")
+
+
+    // Assert that the lists are not empty
+	expect(xValues).toHaveLength(1)
+    expect(yValues ).toHaveLength(1)
+
+    const user = userEvent.setup()
+
+    await user.click(addButton)
+
+    xValues = domTesting.getAllByLabelText(document, "X")
+    yValues = domTesting.getAllByLabelText(document, "Y")
+
+    expect(xValues).toHaveLength(2)
+    expect(yValues).toHaveLength(2)
+})
```

Comment:
The instructions for "Adding values in the chart builder" say 
> Clicking the button should not impact any data the user has already entered.
It looks like we need a second (or expanded) test to verify that behavior.

Reply:
It still looks like we need an additional test here to assert that values entered into the X/Y coordinate inputs are preserved.

Reply:
Do you need help implementing this test?